openhcl/hcl_mapper/src/lib.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Provides a mapper implementation for the page pool that uses the hcl ioctl
+//! crate to map guest memory.
+
+#![cfg(target_os = "linux")]
+#![warn(missing_docs)]
+
+use anyhow::Context;
+use hcl::ioctl::MshvVtlLow;
+use hvdef::HV_PAGE_SIZE;
+use inspect::Inspect;
+use inspect::Response;
+use page_pool_alloc::Mapper;
+use page_pool_alloc::PoolType;
+use sparse_mmap::SparseMapping;
+
+/// A mapper that uses [`MshvVtlLow`] to map pages.
+#[derive(Inspect)]
+#[inspect(extra = "HclMapper::inspect_extra")]
+pub struct HclMapper {
+    #[inspect(skip)]
+    fd: MshvVtlLow,
+}
+
+impl HclMapper {
+    /// Creates a new [`HclMapper`].
+    pub fn new() -> Result<Self, anyhow::Error> {
+        let fd = MshvVtlLow::new().context("failed to open gpa fd")?;
+        Ok(Self { fd })
+    }
+
+    fn inspect_extra(&self, resp: &mut Response<'_>) {
+        resp.field("type", "hcl_mapper");
```

Comment:
this seems like something that would be useful to solve at a more structural level, within Inspect itself... but I suppose this is better than nothing when it comes to identifying the specific kind of mapper you've got.

@jstarks, any thoughts here?